crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -46,6 +47,13 @@ pub(super) struct SemanticIndexBuilder<'db> {
     current_assignments: Vec<CurrentAssignment<'db>>,
     /// The match case we're currently visiting.
     current_match_case: Option<CurrentMatchCase<'db>>,
+
+    /// The [`Unpack`] ingredient for the current definition that belongs to an unpacking
+    /// assignment. This is used to correctly map multiple definitions to the *same* unpacking.
+    /// For example, in `a, b = 1, 2`, both `a` and `b` creates separate definitions but they both
+    /// belong to the same unpacking.
+    current_unpack: Option<Unpack<'db>>,
```

Comment:
Yeah, I initially started with moving this information via `CurrentAssignment` -> `DefinitionNodeRef` -> `DefinitionKind` but that'll involve adding lifetimes and moving them around because `Unpack` uses a different lifetime while `CurrentAssignment` and `DefinitionNodeRef` uses the AST lifetime. But, now that I look at it again I think it might work.

Reply:
Opened https://github.com/astral-sh/ruff/pull/14101